### PR TITLE
Add xml file generation

### LINKED ISSUE #2: Move scripts for generating xml data stream files
The scripts used for generating the xml data stream files should be moved into this repo and removed from the configuration repos, e.g.:

- [generate_xml_datm.py](https://github.com/COSIMA/MOM6-CICE6/blob/1deg_jra55do_iaf/generate_xml_datm.py) 
- [generate_xml_drof.py](https://github.com/COSIMA/MOM6-CICE6/blob/1deg_jra55do_iaf/generate_xml_drof.py)

@ezhilsabareesh8, would you mind doing this since wrote the scripts. I'm happy to review.

### PR DESCRIPTION
**Pull request by ezhilsabareesh8**
Adds generate_xml_datm.py and generate_xml_drof.py  scripts for generating datm and drof input xml files.

Closes #2

### CODE REVIEW

File: xml_file_generation/generate_xml_datm.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# Copyright 2023 ACCESS-NRI and contributors. See the top-level COPYRIGHT file for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Contact: Ezhilsabareesh Kannadasan
```
Comment (dougiesquire) [inline xml_file_generation/generate_xml_datm.py:5]: Could you add a sentence here describing what this script is for?

File: xml_file_generation/generate_xml_drof.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+# Copyright 2023 ACCESS-NRI and contributors. See the top-level COPYRIGHT file for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Contact: Ezhilsabareesh Kannadasan
```
Comment (dougiesquire) [inline xml_file_generation/generate_xml_drof.py:5]: Could you add a sentence describing what this script is for?

**Review by dougiesquire — APPROVED** [review]
Thanks @ezhilsabareesh8. The directory name could be a little more explicit, e.g. `data_stream_xml_generation`

File: xml_file_generation/generate_xml_datm.py (hunk 1 above)
Comment (ezhilsabareesh8) [reply xml_file_generation/generate_xml_datm.py]: Thanks Dougie, added in this commit [48836d1](https://github.com/COSIMA/om3-scripts/commit/48836d1dd84e340b9afcc70f655e58799e3f0b48)

File: xml_file_generation/generate_xml_drof.py (hunk 2 above)
Comment (ezhilsabareesh8) [reply xml_file_generation/generate_xml_drof.py]: Thanks Dougie, added in this commit [05a1fb7](https://github.com/COSIMA/om3-scripts/commit/05a1fb70bf4f00cb668bcc5cb3ff25f00c71261e)

**Comment by ezhilsabareesh8** [comment]
> Thanks @ezhilsabareesh8. The directory name could be a little more explicit, e.g. `data_stream_xml_generation`

Thanks @dougiesquire. I have renamed the directory in this commit [3b0780d](https://github.com/COSIMA/om3-scripts/commit/3b0780d19f3a54797f71028482683d755fdba8ca)

**Pull request merged**